### PR TITLE
Update deps

### PR DESCRIPTION
**Pull request by pocockn**
### Problem

The project has some seriously outdated dependencies for `libp2p`.

The library has changed where things should be imported from and a lot of the main code comes from their core repo.

### Solution

Update all the dependencies and replace the use of deprecated functions throughout the project

### CODE REVIEW

File: blacklist.go [hunk 1]
```diff
@@ -22,8 +22,9 @@ func NewLRUBlacklist(cap int) (*LRUBlacklist, error) {
 }
 
 // Add adds a peer ID
-func (b *LRUBlacklist) Add(p peer.ID) {
+func (b *LRUBlacklist) Add(p peer.ID) bool {
 	b.lru.Add(p, nil)
+	return true
```
Comment (danielcooperxyz) [inline blacklist.go]: I think this should be returning the result of `Add`?

File: p2p.go [hunk 2]
```diff
@@ -413,28 +381,28 @@ func (h *Host) AddUnicastPubSub(topic string, callback HandleUnicast) error {
...
+	t, err := pub.Join(topic)
+	if err != nil {
+		return err
+	}
+
+	sub, err := t.Subscribe()
 	if err != nil {
 		return err
 	}
-	h.pubs[topic] = pub
```
Comment (danielcooperxyz) [inline p2p.go:437]: Is there a specific reason you've removed this? Do we not need to record the publisher?

Comment (pocockn) [reply p2p.go:437]: The pub sub stuff works a bit differently in the latest release. 

You publish on the topic itself so you don't need to store multiple publishers.

```
topics    map[string]*pubsub.Topic
```

And then to publish to that topic you do

```
	t, ok := h.topics[topic]
	if !ok {
		return nil
	}
	return t.Publish(context.Background(), data)
```

Comment (danielcooperxyz) [reply p2p.go:437]: ah, the same as subscribing. ok looks good!

File: p2p.go [hunk 3]
```diff
@@ -11,6 +11,8 @@ import (
 	"os"
 	"time"
 
+	smux "github.com/libp2p/go-libp2p-core/mux"
+	yamux "github.com/libp2p/go-libp2p-yamux"
```
Comment (danielcooperxyz) [inline p2p.go]: can we sort the imports :surfer:

**Review by danielcooperxyz — APPROVED** [review]

**Pull request merged**